### ISSUE TITLE
metadata maintainers is wrong in pyproject.toml

### ISSUE THREAD

**Issue by thebaptiste** [esmf-org/esmf#122]
Hello 

In `https://github.com/esmf-org/esmf/blob/develop/src/addon/esmpy/pyproject.toml`
the metadata `maintainers` is wrong. This leads to problems with recent releases of setuptools.
 `author` should be replaced by `name` like this : 

`maintainers = [ { name = "ESMF Core Team", email = "esmf_support@ucar.edu" } ]`

Reference here  : https://peps.python.org/pep-0621/#authors-maintainers

Regards

**Comment by mboisson** [comment]
I *just* ran into this as well. Error message is: 
```
5025   configuration error: `project.maintainers[{data__maintainers_x}]` must not contain {'author'} properties
5026   GIVEN VALUE:
5027       {
5028           "author": "ESMF Core Team",
5029           "email": "esmf_support@ucar.edu"
5030       }
5031
5032   OFFENDING RULE: 'additionalProperties'
5033
5034   DEFINITION:
5035       {
5036           "$id": "#/definitions/author",
5037           "title": "Author or Maintainer",
5038           "$comment": "https://peps.python.org/pep-0621/#authors-maintainers",
5039           "type": "object",
5040           "additionalProperties": false,
5041           "properties": {
5042               "name": {
5043                   "type": "string",
5044                   "$$description": [
5045                       "MUST be a valid email name, i.e. whatever can be put as a name, before an",
5046                       "email, in :rfc:`822`."
5047                   ]
5048               },
5049               "email": {
5050                   "type": "string",
5051                   "format": "idn-email",
5052                   "description": "MUST be a valid email address"
5053               }
5054           }
5055       }
5056   /tmp/eb-6k3xdk_9/pip-build-env-ld8kaafl/overlay/lib/python3.8/site-packages/setuptools/config/pyprojecttoml.py:108: _BetaConfiguration: Support for `[tool.setuptools]` in `pyproject.toml` is still *beta*.
5057     warnings.warn(msg, _BetaConfiguration)
```

**Comment by mboisson** [comment]
FYI, the patch is simple. I applied this before building ESMF 8.4.0: 
https://github.com/ComputeCanada/easybuild-easyconfigs-installed-avx2/blob/main/2020/ESMF/ESMF-8.4.0-patch-esmpy-pyproj.toml.patch

**Comment by billsacks** [comment]
@thebaptiste and @mboisson - Thank you for reporting this and providing the fix. I reproduced the error and committed the fix. This will be fixed in ESMF 8.5.

**Comment by billsacks** [comment]
This will also be fixed in ESMF 8.4.1, which should be released soon.
